collect_app/src/main/java/org/odk/collect/android/project/ProjectSettingsDialog.kt

```diff
@@ -0,0 +1,57 @@
+package org.odk.collect.android.project
+
+import android.content.Intent
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.button.MaterialButton
+import org.odk.collect.android.R
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.android.preferences.dialogs.AdminPasswordDialogFragment
+import org.odk.collect.android.preferences.screens.AdminPreferencesActivity
+import org.odk.collect.android.preferences.screens.GeneralPreferencesActivity
+import org.odk.collect.android.utilities.AdminPasswordProvider
+import org.odk.collect.android.utilities.DialogUtils
+import javax.inject.Inject
+
+class ProjectSettingsDialog : DialogFragment() {
+
+    @Inject
+    lateinit var adminPasswordProvider: AdminPasswordProvider
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        DaggerUtils.getComponent(context).inject(this)
```

Comment:
This should really be in `onAttach(Context)` in Fragments (mentioned [here](https://developer.android.com/training/dependency-injection/dagger-android#injecting-activities)). Not a big deal but good to fix now.

Reply:
Fixed.